utility/esp_idf/spi.cpp

```diff
@@ -35,38 +34,29 @@ void SPIClass::begin(spi_host_device_t busNo, uint32_t speed)
     busConfig.sclk_io_num = -1; // GPIO14 on SPI2_HOST; GPIO18 on SPI3_HOST
 #endif
     busConfig.quadwp_io_num = -1;
-    busConfig.data2_io_num = -1;
     busConfig.quadhd_io_num = -1;
-    busConfig.data3_io_num = -1;
     busConfig.data4_io_num = -1;
     busConfig.data5_io_num = -1;
     busConfig.data6_io_num = -1;
     busConfig.data7_io_num = -1;
     busConfig.max_transfer_sz = 33; // RF24 lib only buffers 33 bytes for SPI transactions
-    busConfig.flags = 0;
-    busConfig.isr_cpu_id = ESP_INTR_CPU_AFFINITY_AUTO;
-    busConfig.intr_flags = 0;
 
     begin(busNo, speed, SPI_MODE0, &busConfig);
 }
 
 void SPIClass::begin(spi_host_device_t busNo, uint32_t speed, uint8_t mode, spi_bus_config_t* busConfig)
 {
-    esp_err_t ret = spi_bus_initialize(busNo, busConfig, SPI_DMA_CH_AUTO);
+    esp_err_t ret = spi_bus_initialize(busNo, busConfig, SPI_DMA_DISABLED);
```

Comment:
@2bndy5 Lennart has explained his reasoning behind this in depth here https://github.com/tbnobody/OpenDTU/issues/2343

Reply:
Parts of that explanation seem specific to the open DTU fusion's schematic (spanning different versions of that board which uses different variations of ESP32 chips).

My understanding is that `SPI_DMA_DISABLED` should be used on ESP32-S2 when `SPI3_HOST` is passed to the `busNo` parameter. Since `SPI2_HOST` is the parameter's default value, we really only need to beware when using a ESP32-S2.

Should it be configurable with menuconfig or a define (similar to how we'll be configuring the default SPI pins)? This way will allow better identification of the ESP32 variant at CMake build time or earlier (before being compiled).

I'm also inclined to have another optional parameter here because it corresponds to the value of `spi_host_device_t busNo` parameter.

What concerns me is that DMA offers a _significant_ speed-up when it can be used.

Reply:
@2bndy5 No, DMA offers no speed-up in this case. Up to 64 bytes fit directly into the SPI buffer of the ESP32, and copying a maximum of 33 bytes to the buffer of the SPI peripheral is faster than the DMA setup that would have to be done for each transaction separately.

DMA also does not work directly with buffers that are not word-aligned and not a multiple of four long. In such cases, the buffer would internally be copied to a suitable buffer anyways.

On top of that, we use blocking transactions anyways, so the usage of DMA would not "relief" the CPU.

Generally on the ESP32, it usually only makes sense to use DMA when there is a need for transactions longer than the size of the internal SPI buffer (64 bytes).

Reply:
Oh, ok. I was going to ask where you got this information, but then I looked at the docs again (noted under ["Transaction Duration"](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/peripherals/spi_master.html#transaction-duration))

In that case, I'm fine with it being disabled. We don't support Interrupt driven SPI transaction in any other platform anyway (seems more like a quadSPI optimization).